// react-kawaii/lib/native/components/speechBubble/SpeechBubble.js
import React from 'react';
import PropTypes from 'prop-types';
import paths from './paths';
import Face from '../common/face/Face';
import getUniqueId from '../../utils/getUniqueId';
import Wrapper from '../common/wrapper/Wrapper';
import Svg, { G, Path, Use, Defs, Mask } from 'react-native-svg';

var SpeechBubble = function SpeechBubble(_ref) {
  var size = _ref.size,
      color = _ref.color,
      mood = _ref.mood,
      className = _ref.className;
  return React.createElement(Wrapper, {
    className: className
  }, React.createElement(Svg, {
    width: size,
    height: size,
    version: "1.1",
    viewBox: "0 0 134 134",
    xmlns: "http://www.w3.org/2000/svg",
    xmlnsXlink: "http://www.w3.org/1999/xlink"
  }, React.createElement(Defs, null, React.createElement(Path, {
    d: paths.shape,
    id: "kawaii-speechBubble__shape--path"
  }), React.createElement(Path, {
    d: paths.shadow,
    id: "kawaii-speechBubble__shadow--path"
  })), React.createElement(G, {
    id: "Kawaii-speechBubble"
  }, React.createElement(G, {
    id: "Kawaii-speechBubble__body"
  }, React.createElement(Mask, {
    fill: "#fff"
  }, React.createElement(Use, {
    xlinkHref: "#kawaii-speechBubble__shape--path",
    href: "#kawaii-speechBubble__shape--path"
  })), React.createElement(Use, {
    id: "Kawaii-speechBubble__shape",
    fill: color,
    xlinkHref: "#kawaii-speechBubble__shape--path",
    href: "#kawaii-speechBubble__shape--path"
  }), React.createElement(Mask, {
    fill: "#fff"
  }, React.createElement(Use, {
    xlinkHref: "#kawaii-speechBubble__shadow--path",
    href: "#kawaii-speechBubble__shadow--path"
  })), React.createElement(Use, {
    id: "Kawaii-speechBubble__shadow",
    fill: "#000",
    opacity: ".1",
    xlinkHref: "#kawaii-speechBubble__shadow--path",
    href: "#kawaii-speechBubble__shadow--path"
  })), React.createElement(Face, {
    mood: mood,
    transform: "translate(34 57)",
    uniqueId: getUniqueId()
  }))));
};

SpeechBubble.propTypes = {
  /**
   * Size of the width
   * */
  size: PropTypes.number,
  mood: PropTypes.oneOf(['sad', 'shocked', 'happy', 'blissful', 'lovestruck', 'excited', 'ko']),

  /**
   * Hex color
   */
  color: PropTypes.string
};
SpeechBubble.defaultProps = {
  size: 150,
  mood: 'blissful',
  color: '#83D1FB'
};
export default SpeechBubble;
