// react-kawaii/lib/es/components/chocolate/Chocolate.js
import React from 'react';
import PropTypes from 'prop-types';
import paths from './paths';
import getUniqueId from '../../utils/getUniqueId';
import Face from '../common/face/Face';
import Wrapper from '../common/wrapper/Wrapper';

var Chocolate = function Chocolate(_ref) {
  var size = _ref.size,
      color = _ref.color,
      mood = _ref.mood,
      className = _ref.className;
  return React.createElement(Wrapper, {
    className: className
  }, React.createElement("svg", {
    xmlns: "http://www.w3.org/2000/svg",
    width: size * 0.54,
    height: size,
    fill: "none",
    viewBox: "0 0 114 211"
  }, React.createElement("g", {
    id: "chocolate"
  }, React.createElement("path", {
    id: "body",
    fill: "#8E5434",
    d: paths.choco_background
  }), React.createElement("g", {
    id: "bars"
  }, React.createElement("g", {
    id: "choco-inner_1"
  }, React.createElement("path", {
    id: "Union",
    fill: "#fff",
    fillOpacity: "0.1",
    fillRule: "evenodd",
    d: paths.choco_inner_1,
    clipRule: "evenodd"
  })), React.createElement("g", {
    id: "choco-inner_2"
  }, React.createElement("path", {
    id: "Union_2",
    fill: "#fff",
    fillOpacity: "0.1",
    fillRule: "evenodd",
    d: paths.choco_inner_2,
    clipRule: "evenodd"
  })), React.createElement("g", {
    id: "choco-inner_3"
  }, React.createElement("path", {
    id: "Union_3",
    fill: "#fff",
    fillOpacity: "0.1",
    fillRule: "evenodd",
    d: paths.choco_inner_3,
    clipRule: "evenodd"
  })), React.createElement("g", {
    id: "choco-inner_4"
  }, React.createElement("path", {
    id: "Union_4",
    fill: "#fff",
    fillOpacity: "0.1",
    fillRule: "evenodd",
    d: paths.choco_inner_4,
    clipRule: "evenodd"
  })), React.createElement("g", {
    id: "choco-inner_5"
  }, React.createElement("path", {
    id: "Union_5",
    fill: "#fff",
    fillOpacity: "0.1",
    fillRule: "evenodd",
    d: paths.choco_inner_5,
    clipRule: "evenodd"
  })), React.createElement("g", {
    id: "choco-inner_6"
  }, React.createElement("path", {
    id: "Union_6",
    fill: "#fff",
    fillOpacity: "0.1",
    fillRule: "evenodd",
    d: paths.choco_inner_6,
    clipRule: "evenodd"
  }))), React.createElement("g", {
    id: "Group 15"
  }, React.createElement("path", {
    id: "paper",
    fill: color,
    d: paths.paper
  }), React.createElement("g", {
    id: "paper-folded"
  }, React.createElement("path", {
    id: "Vector",
    fill: color,
    d: paths.paper_folded
  }), React.createElement("path", {
    id: "Vector_2",
    fill: "#000",
    fillOpacity: "0.15",
    d: "M0 125.016h102.305L114 92.5 0 125.016z"
  })))), React.createElement(Face, {
    mood: mood,
    transform: "translate(25 150)",
    uniqueId: getUniqueId()
  })));
};

Chocolate.propTypes = {
  /**
   * Size of the width
   * */
  size: PropTypes.number,
  mood: PropTypes.oneOf(['sad', 'shocked', 'happy', 'blissful', 'lovestruck', 'excited', 'ko']),

  /**
   * Hex color
   */
  color: PropTypes.string
};
Chocolate.defaultProps = {
  size: 320,
  mood: 'blissful',
  color: '#fc105c'
};
export default Chocolate;
